instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/__init__.py

```diff
@@ -149,7 +152,8 @@ def _traced_execute_command(func, instance, args, kwargs):
         ) as span:
             if span.is_recording():
                 span.set_attribute(SpanAttributes.DB_STATEMENT, query)
-                _set_connection_attributes(span, instance)
+                if hasattr(instance, "connection_pool"):
```

Comment:
Maybe add the `hasattr` check in the `_set_connection_attributes` function?

Reply:
moved `hasattr` to the `_set_connection_attributes` function.